### PR TITLE
Fix audio format negotiation

### PR DESCRIPTION
**Pull request by wm4**
The first 3 are some old commits.

The second commit of the other 3 commits contains the interesting changes.

### CODE REVIEW

File: DOCS/man/options.rst [hunk 1]
```diff
@@ -4133,6 +4133,20 @@ ALSA audio output options
     or it will work only for files which use the layout implicit to your
     ALSA device).
 
+``--alsa-buffer-time=<microseconds>``
+    Set the requested buffer time in microseconds. A value of 0 skips requesting
+    anything from the ALSA API. This and the period count option uses the ALSA
```
Comment (kevmitch) [inline DOCS/man/options.rst]: should refer explicitly to `alsa-periods`

**Comment by atomnuker** [comment]
Trying to sneak in buffer reduction here as well, huh? The original issues still stand - 25ms is too small to ensure no buffer underruns will happen, increases CPU and is a hack to solve a problem that shouldn't have existed in the first place.

**Comment by wm4** [comment]
> Trying to sneak in buffer reduction here as well, huh? 

No, it's part of another PR, and I was too lazy to switch branches, Mr. Sherlock. Also it's not 25ms anymore.

**Comment by jeeb** [comment]
Some of this PR has already been pushed in, but I have now finished my rounds of testing with the rest:

0. the audio format negotiation rework commit has the first commit's documentation fixups in it. Fixed this already in my [testing branch](https://github.com/jeeb/mpv/commits/pr5716) of this PR.
1. The ALSA/PulseAudio/format negotiation changes seem to work (tested with Fedora 28).
2. With mpv cli the rest seems to work (ditto for Fedora 28).
3. At the scripting and following two dispatch-related commits mpv-android seems to start showing problems.
4. client API changes seem to be OK with mpv-android.

Thus, unless I will find any issues with this, I will be pushing the contents of [this branch](https://github.com/jeeb/mpv/commits/pr5716_without_scripts_or_dispatch) (pr5716 without the scripting/dispatch changes), and after it is found out which side borks mpv-android (quite likely mpv-android itself, but we'll have to check) the rest can be handled as well.

**Comment by jeeb** [comment]
OK, the mpv-android issues were due to it depending on not getting the IDLE event at start (which was an unfortunate misfeature that happened to work during the last year or two).

The new commits look OK, cleaner lavf flushing and making sure libmpv always sends IDLE messages at start.

**Comment by jeeb** [comment]
Aand my usual luck. I had only tested mingw-w64 without the dispatch/script changes. They seem to break mpv.exe, so merging the rest of it and closing this PR.

**Pull request closed without merging**